# HG changeset patch
# User Pantxo <pantxo.diribarne@gmail.com>
# Date 1349275968 14400
#      Wed Oct 03 10:52:48 2012 -0400
# Node ID b95690c001f8b9b93806b650b26f5452db23a5c0
# Parent  cb49817af45730562ae0ebe3ff029ee2c2fc53f4
if bar is given basevalue property, use it

* __bar__.m: Pass base_value to line when creating baseline.

diff --git a/scripts/plot/private/__bar__.m b/scripts/plot/private/__bar__.m
--- a/scripts/plot/private/__bar__.m
+++ b/scripts/plot/private/__bar__.m
@@ -212,17 +212,18 @@ function tmp = bars (ax, vertical, x, y,
                    "cdata", lev, "parent", hg);
       else
         h = patch (yb(:,:,i), xb(:,:,i), "parent", hg);
       endif
     endif
 
     if (i == 1)
       x_axis_range = get (ax, "xlim");
-      h_baseline = line (x_axis_range, [0, 0], "color", [0, 0, 0]);
+      h_baseline = line (x_axis_range, [base_value, base_value],
+                         "color", [0, 0, 0]);
       set (h_baseline, "handlevisibility", "off");
       set (h_baseline, "xliminclude", "off");
       addlistener (ax, "xlim", @update_xlim);
       addlistener (h_baseline, "ydata", @update_baseline);
       addlistener (h_baseline, "visible", @update_baseline);
     endif
 
     ## Setup the hggroup and listeners
